nemo_aligner/data/nlp/datasets.py

```diff
@@ -323,13 +327,24 @@ def __getitem__(self, idx):
             torch.LongTensor(reject_labels), (0, max_curr_seq_len - len(reject_labels)), mode="constant", value=-100
         )
 
+        # ignore the example whose tokenized text exceeds max seq length.
+        if max_curr_seq_len > self.seq_length:
+            chosen_tokens = chosen_tokens[: self.seq_length]
+            rejected_tokens = rejected_tokens[: self.seq_length]
```

Comment:
Doing `chosen_tokens[: self.seq_length]` is not a good approach, because you're taking a very large amount of tokens to be used/calculated in the forward pass later on in `get_forward_output_and_loss_func`, but then the loss will be zeroed out anyway. Imagine what happens in a long context model where `seq_length` could be like 64k. I recommend you use a smaller, fixed number instead like doing:

```
chosen_tokens = chosen_tokens [:32]
rejected_tokens = rejected_tokens [:32]
```

And then shifting everything else to conform to length=32

Reply:
i think if we filter this out in init then we don't even need this logic

Reply:
Updated a [commit](https://github.com/NVIDIA/NeMo-Aligner/pull/195/commits/d577b176e2e9f20e546ca1ef044f5679f5ae9bbc) with changing `self.seq_length` to `32` for efficiency. I suggest that we leave the `init filtering` to a separate PR since it might need more careful discussions.